misc/python/materialize/mzbuild.py

```diff
@@ -671,18 +671,27 @@ def build(self) -> None:
         ]
         spawn.runv(cmd, stdin=f, stdout=sys.stderr.buffer)
 
-    def try_pull(self) -> bool:
+    def try_pull(self, max_retries: int) -> bool:
         """Download the image if it does not exist locally. Returns whether it was found."""
         ui.header(f"Acquiring {self.spec()}")
         if not self.acquired:
-            try:
-                spawn.runv(
-                    ["docker", "pull", self.spec()],
-                    stdout=sys.stderr.buffer,
-                )
-                self.acquired = True
-            except subprocess.CalledProcessError:
-                pass
+            for retry in range(1, max_retries + 1):
+                try:
+                    spawn.runv(
+                        ["docker", "pull", "x" + self.spec()],
+                        stdout=sys.stderr.buffer,
+                    )
+                    self.acquired = True
+                except subprocess.CalledProcessError:
+                    if retry < max_retries:
+                        # There seems to be no good way to tell what error
```

Comment:
I'm not happy with this. We are basically trading CI flakiness for wasted pulls (can take a second). I couldn't find a better solution, if anyone has a good idea tell me.

Reply:
Retry only if running in a CI context, Buildkite sets both `BUILDKITE="true"` and `CI="true"` 
`